lib/srv/db/sqlserver/connect.go

```diff
@@ -64,20 +68,42 @@ func (c *connector) Connect(ctx context.Context, sessionCtx *common.Session, log
 		TypeFlags:    loginPacket.TypeFlags(),
 	}
 
-	auth, err := c.getAuth(sessionCtx)
-	if err != nil {
-		return nil, nil, trace.Wrap(err)
-	}
-
-	connector := mssql.NewConnectorConfig(msdsn.Config{
+	dsnConfig := msdsn.Config{
 		Host:         host,
 		Port:         portI,
 		Database:     sessionCtx.DatabaseName,
 		LoginOptions: options,
 		Encryption:   msdsn.EncryptionRequired,
 		TLSConfig:    tlsConfig,
 		PacketSize:   loginPacket.PacketSize(),
-	}, auth)
+	}
+
+	var connector *mssql.Connector
+	switch {
+	case sessionCtx.Database.IsAzure() && sessionCtx.Database.GetAD().IsEmpty():
+		// If the client is connecting to Azure SQL, and no AD configuration is
+		// provided, authenticate using the Azure AD Integrated authentication
+		// method.
+		managedIdentityID, err := c.Auth.GetAzureIdentityResourceID(ctx, sessionCtx.DatabaseUser)
+		if err != nil {
+			return nil, nil, trace.Wrap(err)
+		}
+
+		connector, err = azuread.NewConnectorFromConfig(dsnConfig, map[string]string{
+			"fedauth":     azuread.ActiveDirectoryManagedIdentity,
+			"resource id": managedIdentityID,
+		})
+		if err != nil {
+			return nil, nil, trace.Wrap(err)
+		}
+	default:
+		auth, err := c.getAuth(sessionCtx)
+		if err != nil {
+			return nil, nil, trace.Wrap(err)
+		}
+
+		connector = mssql.NewConnectorConfig(dsnConfig, auth)
+	}
```

Comment:
Can this be made into simple `if-else`? For example:

```suggestion
        var connector *mssql.Connector
	
	if sessionCtx.Database.IsAzure() && sessionCtx.Database.GetAD().IsEmpty() {
		// If the client is connecting to Azure SQL, and no AD configuration is
		// provided, authenticate using the Azure AD Integrated authentication
		// method.
		managedIdentityID, err := c.Auth.GetAzureIdentityResourceID(ctx, sessionCtx.DatabaseUser)
		if err != nil {
			return nil, nil, trace.Wrap(err)
		}
		connector, err = azuread.NewConnectorFromConfig(dsnConfig, map[string]string{
			"fedauth":     azuread.ActiveDirectoryManagedIdentity,
			"resource id": managedIdentityID,
		})
		if err != nil {
			return nil, nil, trace.Wrap(err)
		}
	} else {
		auth, err := c.getAuth(sessionCtx)
		if err != nil {
			return nil, nil, trace.Wrap(err)
		}

		connector = mssql.NewConnectorConfig(dsnConfig, auth)
	}
```